src/alcotest.mli

```diff
@@ -74,11 +77,28 @@ val run_with_args :
   string ->
   'a Cmdliner.Term.t ->
   'a test list ->
-  unit
+  return
 (** [run_with_args n a t] Similar to [run a t] but take an extra
     argument [a]. Every test function will receive as arguement the
     evaluation of the [Cdmliner] term [a]: this is useful to configure
     the test behaviors using the CLI. *)
+end
+
+include TESTER with type return = unit
+
+module type MONAD = sig
+  type 'a t
+
+  val return : 'a -> 'a t
+
+  val bind : 'a t -> ('a -> 'b t) -> 'b t
+end
+
+module MonadicTester (M : MONAD) : TESTER with type return = unit M.t
```

Comment:
It's best to avoid camel-case (especially in the API). I guess here just call it `Make (M: MONAD)`